app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt

```diff
@@ -33,38 +36,44 @@ class SkillsListAdapter(
+            } else {
+                holder.skillName.text = skillData.skillName
+            }
 
-        if (skillData.author.isEmpty() || skillData.author.equals(DEFAULT_AUTHOR)) {
-            holder.skillAuthorName.text = context.getString(R.string.no_skill_author)
-        } else {
-            holder.skillAuthorName.text = skillData.author
-        }
+            if (skillData.author.isEmpty() || skillData.author.equals(DEFAULT_AUTHOR)) {
+                holder.skillAuthorName.text = context.getString(R.string.no_skill_author)
+            } else {
+                holder.skillAuthorName.text = skillData.author
+            }
 
-        if (skillData.examples.isEmpty() || skillData.examples.contains(DEFAULT_EXAMPLE))
-            holder.skillExample.text = ""
-        else
-            holder.skillExample.text = StringBuilder("\"").append(skillData.examples[0]).append("\"")
+            if (skillData.examples.isEmpty() || skillData.examples.contains(DEFAULT_EXAMPLE))
+                holder.skillExample.text = ""
+            else
+                holder.skillExample.text = StringBuilder("\"").append(skillData.examples[0]).append("\"")
 
-        if (skillData.image.isEmpty()) {
-            holder.skillImage.setImageResource(R.drawable.ic_susi)
-        } else {
-            val imageUrl: String = skillDetails.skillsList.get(position).group.replace(" ", "%20") + "/en/" + skillData.image
-            Picasso.get().load(StringBuilder(imageLink)
-                .append(imageUrl).toString())
-                .fit().centerCrop()
-                .error(R.drawable.ic_susi)
-                .transform(CircleTransform())
-                .into(holder.skillImage)
-        }
-        val stars = skillData.skillRating?.stars
-        if (stars != null) {
-            holder.skillRating.rating = stars.averageStar
-            holder.skillTotalRatings.text = stars.totalStar.toString()
+            if (skillData.image.isEmpty()) {
+                holder.skillImage.setImageResource(R.drawable.ic_susi)
+            } else {
+                val imageUrl: String = skillDetails.skillsList.get(position).group.replace(" ", "%20") + "/en/" + skillData.image
+                Picasso.get().load(StringBuilder(imageLink)
+                        .append(imageUrl).toString())
+                        .fit().centerCrop()
+                        .error(R.drawable.ic_susi)
+                        .transform(CircleTransform())
+                        .into(holder.skillImage)
+            }
+            val stars = skillData.skillRating?.stars
+            if (stars != null) {
+                holder.skillRating.rating = stars.averageStar
+                holder.skillTotalRatings.text = stars.totalStar.toString()
+            }
+        } catch (illegalArgument: IllegalArgumentException) {
+            Timber.e(illegalArgument)
+        } catch (nullPointer: NullPointerException) {
```

Comment:
Don't catch NPE

Reply:
**Logcat of the Crash**
java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
        at org.fossasia.susi.ai.skills.groupwiseskills.adapters.recycleradapters.SkillsListAdapter.onBindViewHolder(SkillsListAdapter.kt:39)
        at org.fossasia.susi.ai.skills.groupwiseskills.adapters.recycleradapters.SkillsListAdapter.onBindViewHolder(SkillsListAdapter.kt:23)
        at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6781)
        at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6823)
        at android.support.v7.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5752)
        at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6019)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5858)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5854)
        at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2230)
        at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1557)
        at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1517)
        at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:612)
        at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3924)
        at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3641)
        at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:4194)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.support.v4.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:625)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.support.v7.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:444)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1959)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1813)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1722)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2484)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2181)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1293)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6599)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:800)
        at android.view.Choreographer.doCallbacks(Choreographer.java:603)
    	at android.view.Choreographe

Reply:
Yeah, so add null check